lib/board.rb

```diff
@@ -0,0 +1,137 @@
+class Board
+  attr_reader :cells
+
+  def initialize
+    @cells = {
+      'A1' => Cell.new('A1'),
+      'A2' => Cell.new('A2'),
+      'A3' => Cell.new('A3'),
+      'A4' => Cell.new('A4'),
+      'B1' => Cell.new('B1'),
+      'B2' => Cell.new('B2'),
+      'B3' => Cell.new('B3'),
+      'B4' => Cell.new('B4'),
+      'C1' => Cell.new('C1'),
+      'C2' => Cell.new('C2'),
+      'C3' => Cell.new('C3'),
+      'C4' => Cell.new('C4'),
+      'D1' => Cell.new('D1'),
+      'D2' => Cell.new('D2'),
+      'D3' => Cell.new('D3'),
+      'D4' => Cell.new('D4')
+    }
+  end
+
+  def valid_coordinate?(cell_coordinate)
+    @cells.include?(cell_coordinate)
+  end
+
+  def valid_placement?(ship, coords)
+    coords.each do |coordinate|
+      if !(@cells[coordinate].empty?)
+        return false
+      end
+    end
+
+    if ship.length == 2
+      same_length?(ship, coords) && (consecutive_submarine_on_same_letter?(coords) || consecutive_submarine_on_same_number?(coords))
+    elsif ship.length == 3
+      same_length?(ship, coords) && (consecutive_cruiser_on_same_letter?(coords) || consecutive_cruiser_on_same_number?(coords))
+    end
+  end
+
+  def same_length?(ship, coords)
+    coords.count == ship.length
+  end
+
+  def consecutive_submarine_on_same_letter?(coords)
```

Comment:
Here we added the methods I used for my board. We replaced her valid_placement method and everythin worked just fine still, which was real nice. Phew